--- java/org/apache/jasper/compiler/JDTCompiler.java	2014-09-26 11:25:17.000000000 +0200
+++ java/org/apache/jasper/compiler/JDTCompiler.java	2014-10-29 17:08:42.268639713 +0100
@@ -166,7 +166,7 @@
                 return result;
             }
 
-            @Override
+            //@Override
             public boolean ignoreOptionalProblems() {
                 return false;
             }
@@ -342,9 +342,9 @@
             } else if(opt.equals("1.7")) {
                 settings.put(CompilerOptions.OPTION_Source,
                              CompilerOptions.VERSION_1_7);
-            } else if(opt.equals("1.8")) {
-                settings.put(CompilerOptions.OPTION_Source,
-                             CompilerOptions.VERSION_1_8);
+            //} else if(opt.equals("1.8")) {
+            //    settings.put(CompilerOptions.OPTION_Source,
+            //                CompilerOptions.VERSION_1_8);
             } else {
                 log.warn("Unknown source VM " + opt + " ignored.");
                 settings.put(CompilerOptions.OPTION_Source,
@@ -386,11 +386,11 @@
                              CompilerOptions.VERSION_1_7);
                 settings.put(CompilerOptions.OPTION_Compliance,
                         CompilerOptions.VERSION_1_7);
-            } else if(opt.equals("1.8")) {
-                settings.put(CompilerOptions.OPTION_TargetPlatform,
-                             CompilerOptions.VERSION_1_8);
-                settings.put(CompilerOptions.OPTION_Compliance,
-                        CompilerOptions.VERSION_1_8);
+            //} else if(opt.equals("1.8")) {
+            //    settings.put(CompilerOptions.OPTION_TargetPlatform,
+            //                 CompilerOptions.VERSION_1_8);
+            //    settings.put(CompilerOptions.OPTION_Compliance,
+            //            CompilerOptions.VERSION_1_8);
             } else {
                 log.warn("Unknown target VM " + opt + " ignored.");
                 settings.put(CompilerOptions.OPTION_TargetPlatform,
